docs/notebooks/util/plotting.py

```diff
@@ -208,6 +209,101 @@ def plot_bo_points(
             ax.scatter(pts[i, 0], pts[i, 1], obs_values[i], c=col_pts[i], marker=mark_pts[i])
 
 
+def plot_bo_points_in_obj_space(
+    obs_values,
+    num_init=None,
+    mask_fail=None,
+    figsize=None,
+    xlabel="Obj 1",
+    ylabel="Obj 2",
+    zlabel="Obj 3",
+    title=None,
+    m_init="x",
+    m_add="o",
+    c_pass="tab:green",
+    c_fail="tab:red",
+    c_pareto="tab:purple",
+    only_plot_pareto=False,
+):
+    """
+    Adds scatter points in objective space, used for multi-objective optimization (2 objective only).
+    Markers and colors are chosen according to BO factors.
+    :param obs_values:
+    :param num_init: initial number of BO points
+    :param mask_fail: Boolean vector, True if the corresponding observation violates the constraint(s)
+    :param title:
+    :param xlabel:
+    :param ylabel:
+    :param figsize:
+    :param only_plot_pareto: if set true, only plot the pareto points
+    """
+    obj_num = obs_values.shape[-1]
+    tf.debugging.assert_shapes([])
+    assert obj_num == 2 or obj_num == 3, NotImplementedError(
+        "Only support 2/3-objective" " function plot but found: {}".format(obj_num)
+    )
+
+    _, dom = non_dominated(obs_values)
+    idx_pareto = (
+        np.where(dom == 0) if mask_fail is None else np.where(np.logical_and(dom == 0, ~mask_fail))
+    )
+
+    pts = obs_values
+    num_pts = pts.shape[0]
+
+    col_pts, mark_pts = format_point_markers(
+        num_pts, num_init, idx_pareto, mask_fail, m_init, m_add, c_pass, c_fail, c_pareto
+    )
+    if only_plot_pareto:
+        col_pts = col_pts[idx_pareto]
+        mark_pts = mark_pts[idx_pareto]
+        pts = pts[idx_pareto]
+
+    if obj_num == 2:
+        fig, ax = plt.subplots(figsize=figsize)
+    else:
+        fig = plt.figure()
+        ax = fig.add_subplot(111, projection="3d")
+    for i in range(pts.shape[0]):
+        ax.scatter(*pts[i], c=col_pts[i], marker=mark_pts[i])
+    ax.set_xlabel(xlabel)
+    ax.set_ylabel(ylabel)
+    if obj_num == 3:
+        ax.set_zlabel(zlabel)
+    if title is not None:
+        ax.set_title(title)
+    return fig, ax
+
+
+def plot_mo_history(
+    obs_values,
+    metric_func,
+    num_init=None,
+    mask_fail=None,
+    figsize=None,
+):
+    """
+    Draw the performance measure for multi-objective optimization
+    :param obs_values:
+    :param metric_func: a callable function calculate metric score
+                        metric = measure_func(observations)
+    :param num_init:
+    :param mask_fail:
+    :param figsize
+    """
```

Comment:
Do we currently use any other metrics than the one from the notebook?

Reply:
We also have HV, which is something already [there](https://github.com/secondmind-labs/trieste/blob/d9b81a04c09852c882b38f0cd0c5874e5783acb3/trieste/utils/pareto.py#L280) and can also directly plot in our notebook, but for other metrics like IGD not yet implemented (guess IGD is not often used in bo community, CMIIW).